### PR TITLE
Device and tasks

### PR DESCRIPTION
**Pull request by Darkros1245**
Created tasks and device endpoints including tests for it for the back end and front end

### CODE REVIEW

File: frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt [hunk 1]
```diff
@@ -102,6 +102,75 @@ class ApiServiceTest {
         }
     }
```
Comment (S3JER) [inline frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt:104]: Missing @Test

File: backend/src/main.rs [hunk 2]
```diff
@@ -252,124 +359,98 @@ mod tests {
...
+        let all_tasks = get_tasks(&mut app, auth_token).await;
+        let expected: Vec<Task> = Vec::new();
 
-        let response = ServiceExt::<Request<Body>>::ready(&mut app)
-            .await
-            .unwrap()
-            .call(request)
-            .await
-            .unwrap();
+        assert_eq!(all_tasks, expected);
```
Comment (mads256h) [inline backend/src/main.rs:407]:
```suggestion
        assert!(all_tasks.is_empty());
```

File: backend/src/main.rs [hunk 3]
```diff
@@ -252,124 +359,98 @@ mod tests {
...
+        let auth_token = get_account(&mut app).await;
+        let auth_token = auth_token_to_uuid(auth_token);
+        let device = generate_device(&mut app, auth_token.clone(), 1000.0).await;
+        delete_device(&mut app, auth_token.clone(), device).await;
+
+        let all_devices = get_devices(&mut app, auth_token).await;
+        let expected: Vec<Device> = Vec::new();
 
-        assert_eq!(all_tasks.first().unwrap(), &created_task);
+        assert_eq!(all_devices, expected);
```
Comment (mads256h) [inline backend/src/main.rs:453]:
```suggestion
        assert!(all_devices.is_empty());
```

File: frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt [hunk 4]
```diff
@@ -102,6 +102,76 @@ class ApiServiceTest {
...
+
+            assert(deleteResponse.isSuccessful) { printErrorContext(deleteResponse) }
+
+            val getTasksResponse = apiService.getTasks(authToken.toString())
+
+            assert(getTasksResponse.isSuccessful) { printErrorContext(getTasksResponse) }
+
+            val tasks = getTasksResponse.body()!!
+
+            assert(tasks.isEmpty())
```
Comment (mads256h) [inline frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt:124]: Nitpick
```suggestion
            val (authToken, _) = createAccount("get_all_tasks_test")
            val device = createDevice(authToken)
            val task = createTask(authToken, device)

            val deleteResponse = apiService.deleteTask(authToken.toString(), task.id)
            assert(deleteResponse.isSuccessful) { printErrorContext(deleteResponse) }

            val getTasksResponse = apiService.getTasks(authToken.toString())
            assert(getTasksResponse.isSuccessful) { printErrorContext(getTasksResponse) }

            val tasks = getTasksResponse.body()!!
            assert(tasks.isEmpty())
```

File: frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt [hunk 5]
```diff
@@ -102,6 +102,76 @@ class ApiServiceTest {
...
+
+            val device = createDevice(authToken)
+
+            val response = apiService.getDevices(authToken.toString())
+
+            assert(response.isSuccessful) { printErrorContext(response) }
+
+            val devices = response.body()!!
+
+            assert(devices.size == 1)
```
Comment (mads256h) [inline frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt:150]: Nitpick
```suggestion
            val (authToken, _) = createAccount("get_all_tasks_test")
            val device = createDevice(authToken)

            val response = apiService.getDevices(authToken.toString())
            assert(response.isSuccessful) { printErrorContext(response) }

            val devices = response.body()!!
            assert(devices.size == 1)
```

File: frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt [hunk 6]
```diff
@@ -102,6 +102,76 @@ class ApiServiceTest {
...
+
+            assert(deleteResponse.isSuccessful) { printErrorContext(deleteResponse) }
+
+            val getTasksResponse = apiService.getDevices(authToken.toString())
+
+            assert(getTasksResponse.isSuccessful) { printErrorContext(getTasksResponse) }
+
+            val tasks = getTasksResponse.body()!!
+
+            assert(tasks.isEmpty())
```
Comment (mads256h) [inline frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt:171]:
```suggestion
            val (authToken, _) = createAccount("get_all_tasks_test")
            val device = createDevice(authToken)

            val deleteResponse = apiService.deleteDevice(authToken.toString(), device.id)
            assert(deleteResponse.isSuccessful) { printErrorContext(deleteResponse) }

            val getTasksResponse = apiService.getDevices(authToken.toString())
            assert(getTasksResponse.isSuccessful) { printErrorContext(getTasksResponse) }

            val tasks = getTasksResponse.body()!!
            assert(tasks.isEmpty())
```

**Review by mads256h — CHANGES_REQUESTED** [review]
Small issues

**Review by mads256h — APPROVED** [review]

**Pull request merged**